### DISCUSSION TITLE
Adding Column to Frontend Composer table where the value is based on other column values

### DISCUSSION THREAD

**Discussion by rmuller-vu** [mia-platform/community#548] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.102.8

### Description

We are trying to add a column to our Frontend Composer table where the value is shows if 2 other columns are null/empty. Below is the config. We have tried to use the value and `args.[1].approvedOn` but it doesn't seem to have knowledge of the approvedOn value in the same row. How can we use values from other columns in a row to drive the value of our new column?

The column we want to do custom logic in is called `needsAttention`

```json
{
  "definitions": {
    "dataSchema": {
      "type": "object",
      "required": [
        "projectId",
        "versionName",
        "versionCreatedAt"
      ],
      "properties": {
        "needsAttention": {
          "type": "object",
          "visualizationOptions": {
            "template": "{{#unless approvedOn}}A Value Here{{/unless}}",
            "hidden": false,
            "hiddenLabel": true
          }
        },
        "approvedOn": {
          "type": "string",
          "label": "Approved On",
          "description": "Date and time the version was approved on ",
          "format": "date-time",
          "formOptions": {
            "readOnlyOnUpdate": true,
            "hidden": false,
            "hiddenOnInsert": true,
            "hiddenOnUpdate": false
          }
        },
        "deniedOn": {
          "type": "string",
          "label": "Denied On",
          "description": "Date and time the version was denied on",
          "format": "date-time",
          "formOptions": {
            "readOnlyOnUpdate": true,
            "hidden": false,
            "hiddenOnInsert": true,
            "hiddenOnUpdate": false
          }
        }
      }
    }
  }
}
```

### Actual Outcome

The template `#unless approvedOn` or `#unless args.[1].approvedOn` always equates to true even when values exist in the `approvedOn` column

### Expected Outcome

If `approvedOn` has a value then `needsAttention` should be empty. If `approvedOn` is null or empty then `needsAttention` should have `A Value Here` show up in the column

**Comment by LiliumCandidum** [comment, marked as answer]
Hi @rmuller-vu,

`args.[1].approvedOn` doesn't work in `template` with object fields because you can only use it to access the current field value with the `value` keyword.

However you can override the table cell using a div:
```json
"needsAttention": {
  "type": "object",
  "visualizationOptions": {
    "tag": "div",
    "properties": {
      "textContent": "{{#unless args.[1].orderedAt}}A Value Here{{/unless}}"
    },
    "hidden": false,
    "hiddenLabel": true
  }
},
```

**Reply by rmuller-vu** [reply]
This worked. I do have another question. If we have an object field we are able to use `value` to access data from the current row? So, I could use `value.approvedOn` and get the value of that field for the current row?

**Reply by LiliumCandidum** [reply]
No you can use `value` only to access the object field value in the current row.
For example if `needsAttention` in the current row contains `{ key1: value1, key2: value2 }`, you can access `key1` with `"template": "{{value.key1}}"` and it will show `value1`

**Reply by rmuller-vu** [reply]
Okay, that makes sense. Thank you!
